src/cli/cluster/cluster_manager.js

```diff
@@ -95,26 +91,26 @@ export default class ClusterManager {
     if (opts.watch) {
       const pluginPaths = config.get('plugins.paths');
       const scanDirs = config.get('plugins.scanDirs');
-      const extraPaths = [
-        ...pluginPaths,
-        ...scanDirs,
-      ];
+      const extraPaths = [...pluginPaths, ...scanDirs];
 
       const extraIgnores = scanDirs
         .map(scanDir => resolve(scanDir, '*'))
         .concat(pluginPaths)
-        .reduce((acc, path) => acc.concat(
-          resolve(path, 'test'),
-          resolve(path, 'build'),
-          resolve(path, 'target'),
-          resolve(path, 'scripts'),
-          resolve(path, 'docs'),
-        ), []);
+        .reduce(
+          (acc, path) =>
+            acc.concat(
+              resolve(path, 'test'),
+              resolve(path, 'build'),
+              resolve(path, 'target'),
+              resolve(path, 'scripts'),
+              resolve(path, 'docs'),
+              resolve(path, 'x-pack/plugins/canvas/canvas_plugin_src') // prevents server from restarting twice for Canvas plugin changes
```

Comment:
We should add paths under `canvas_src` that only contain UI code, but we can do that in a followup pull.